src/workerd/api/streams/common.h

```diff
@@ -31,6 +31,31 @@ class TransformStreamDefaultController;
 
 using rpc::StreamEncoding;
 
+enum class ReadAllTextOption : uint8_t {
+  NONE = 0,
+  NULL_TERMINATE = 1 << 0,
+  STRIP_BOM = 1 << 1,
+};
```

Comment:
My other comment got buried a bit... why use an option for this at all? Why not just check for the presence of the compat flag directly in the read all impls?